src/main/java/org/tron/core/db/Manager.java

```diff
@@ -844,6 +862,26 @@ public synchronized void pushBlock(final BlockCapsule block)
     long start = System.currentTimeMillis();
     try (PendingManager pm = new PendingManager(this)) {
 
+      Future<Boolean> future = executorService.submit(() -> {
+        try {
+          if (!block.generatedByMyself) {
+            preValidateTransactionSign(block);
+          }
+        } catch (Exception e) {
+          logger.error("preValidateTransactionSign fail! block info: {}", block.toString(),
+              e.getMessage());
+          return false;
+        }
+        return true;
+      });
+      while (!done) {
+        try {
+          Thread.sleep(1);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Replace the call to "Thread.sleep(...)" with a call to "wait(...)". ([squid:S2276](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2276))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1916&issues=AWgnKDa4rt4Kct_4Nf2H&open=AWgnKDa4rt4Kct_4Nf2H)

[ISSUE_KEY]: AWgnKDa4rt4Kct_4Nf2H